data/data/bootstrap/bootstrap-in-place/usr/local/bin/installer-master-bootstrap-in-place-gather.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
```

Comment:
@staebler @crawford PTAL

Reply:
https://docs.google.com/presentation/d/1nEOaXePYcRRF-6U0X14s0BzuFYfpMzqmSVP7dX5rFqg/edit#slide=id.g3d946fe0a1_0_0

Reply:
This will be called by the installer-gather.
This script will be used as the installer-gather post-reboot, see [here](https://github.com/openshift/installer/pull/4482/files#diff-d09d8f9e83a054002d5344223d496781ea603da7c52706dfcf33debf8ceb1df3R47)

Reply:
Thanks for pointing out where it is dropped on disk, that clarifies a lot. I'm concerned about the installer-bootstrap-in-place-gather.sh diverging from installer-gather.sh and us not gathering updated information from the cluster post-reboot. For example, one section[1] in installer-gather.sh does `Gathering cluster resources ...` Because this only ran on the bootstrap host pre-reboot, it is what will land in the log bundle and may be very confusing to someone trying to diagnose. Are we assuming that if this script can connect using the oc cli, then the user should instead be using `oc adm must-gather`?

https://github.com/openshift/installer/blob/master/data/data/bootstrap/files/usr/local/bin/installer-gather.sh#L82

Reply:
With regards to what information it is collecting - it is not diverging.

For getting bootstrap logs, it simply uses the bootstrap log bundle that was collected before reboot by the original `installer-gather.sh` pre-pivot, passed on to it via the master ignition.

For getting master logs, it simply calls `installer-masters-gather.sh` on itself - which is exactly what the originals `installer-gather.sh` script runs on remote masters.

Calling `openshift-installer gather bootstrap` after the bootstrap node is taken down in non-single-node deployments is not possible. In single node deployments, we added that "ability" that was not there before. The user can still call `oc adm must-gather`, but if the user also wants information about the bootstrap process, they still have the option to call `openshift-installer gather bootstrap`.

Reply:
I'm not concerned about the `installer-masters-gather.sh` portion of this. The log bundle will clearly contain the logs from the host from a control plane perspective. I am concerned that the various `oc get`[1] (and similar) portions of the `installer-gather.sh` script are only executed while the host is configured as a pre-reboot bootstrap host. This point-in-time snapshot of the API does not contain an up-to-date picture of the cluster status at the time the `openshift-installer gather bootstrap` command is executed. There will surely be changes to the cluster post-reboot. This unnecessarily complicates the troubleshooting process by removing a piece of the puzzle.

To this same point, perhaps the gather scripts should be removed from the host once the cluster passes the requirements where a non-single-node installation would tear down the bootstrap node. A "day 2" `gather bootstrap` log bundle is going to be equally confusing to troubleshoot. This will reinforce the use of `must-gather` at this point.

[1] https://github.com/openshift/installer/blob/95855a1b77e1a6904a91e4b4a1f931d194f659f2/data/data/bootstrap/files/usr/local/bin/installer-gather.sh#L82-L111

Reply:
Agree, it might be a bit misleading.
We can place the pre-reboot logs under bootstrap-pre-reboot directory.
We can also add the `oc get` commands to installer-master-bootstrap-in-place-gather.sh
That been said, the bootstrap log bandle and the installer-master-bootstrap-in-place-gather.sh should be deleted upon successful installation.

Reply:
Added [PR](https://github.com/eranco74/installer/pull/11) in response to this discussion, @jstuever and @eranco74 please review

Reply:
That PR might be our solution. However, before that I'm interested if we can simply run the original `installer-gather.sh` on the node, which should run the `installer-masters-gather.sh` on itself. We could then update`installer-gather.sh` to override specific subfolders (bootstrap, rendered-assets, unit-assets) when the BOOTSTRAP_PHASE_LOG_BUNDLE_ARCHIVE_PATH file exists and then dump everything else into a new subfolder. Alternatively, we can instead just dump everything in this file into the new subfolder.

Reply:
That makes sense and sounds like it will simplify things, I'll take a look

Reply:
Got rid of the PR mentioned above.

Created this instead https://github.com/eranco74/installer/pull/12

Is that what you had in mind?